### ISSUE TITLE
UI displays commit hash even when display name is available for version

### ISSUE THREAD

**Issue by luispollo** [spinnaker/keel#1390]
Given this API response:
```json
{
  "applicationPaused": false,
  "hasManagedResources": false,
  "currentEnvironmentConstraints": [],
  "artifacts": [
    {
      "name": "@netflix-treasure/treasure-example-ui",
      "type": "npm",
      "reference": "treasure-example",
      "versions": [
        {
          "version": "1.0.0-10-5d44dc4890536dbb9888b6b6e22f3250734db082",
          "displayName": "1.0.0",
          "environments": [],
          "build": {
            "id": 10
          },
          "git": {
            "commit": "5d44dc4890536dbb9888b6b6e22f3250734db082"
          }
        }
      ]
    }
  ]
}
```
The UI will render the version like so:
![image](https://user-images.githubusercontent.com/1323478/88342704-bf317f00-ccf4-11ea-861c-816e834bca33.png)

**Comment by luispollo** [comment]
@erikmunson @vigneshm FYI

**Comment by erikmunson** [comment]
This is functioning as desired for now — the [overall goal is to favor structured information](https://github.com/spinnaker/keel/issues/887#issuecomment-600378722) that's common across artifacts and fall back to `displayName` when we're unable to use structured info for some reason (e.g. for docker images today that don't adhere to a tagging pattern we can pull git/build data out of).

I do have an issue open for [adding a structured semantic version field](https://github.com/spinnaker/keel/issues/1068) and using it where appropriate — if you're interested in implementing that (ideally across all the current artifact types + the common plugin contract) I'd be happy to experiment with using it in the way I proposed in that GH issue (annotated screenshots). Note that at the time we weren't super confident that it would be the right way to show versions for everyone — we got some early feedback from a couple teams that semantic version is helpful, but there are entire groups of internal users that don't use it at all (all of them on docker AFAICT). We'll need to figure out the appropriate fallbacks and where they apply, which probably won't be right the first couple of tries.

Going to close this specific issue in favor of #1068
